schema/application-descriptor-schema.json

```
@@ -0,0 +1,184 @@
+                        "title": "The language processor used for the source group.",
+                        "description": "The language processor field is for reference only and not yet processed by any automation tasks."
+                    },
+                    "fileExtension": {
+                        "type": "string",
+                        "title": "The file extension used for the source group.",
+                        "description": "The file extension of the files."
+                    },
+                    "artifactsType": {
+                        "type": "string",
+                        "title": "The type of artifacts in the source group.",
+                        "description": "The type of artifacts is important in the automation tasks to understand the relationships.",
+                        "enum": [
+                            "Program",
+                            "Include File"
+                        ]
+                    },
+                    "files": {
+                        "type": "array",
+                        "title": "The list of files in the source group",
+                        "description": "Array of files that belong and are actively managed within the application's git repository.",
+                        "items": {
+                            "type": "object",
+                            "properties": {
+                                "name": {
+                                    "type": "string",
+                                    "title": "The name of the file.",
+                                    "description": "Name of the file w/o file extension. Capitalization has to meet how the file is stored in the repository."
+                                },
+                                "type": {
+                                    "type": "string",
+                                    "title": "The type of the file",
+                                    "description": "The former build type used to generate build properties. This field is not used in automation tasks."
+                                },
+                                "usage": {
+                                    "type": "string",
+                                    "title": "The usage of the file.",
+                                    "description": "The usage of the file impacts how build outputs for the file are packaged. Case-sensitive. \nInclude files with usage set to 'public' or 'shared' can be published via the application tar file and can then be referenced by other application teams. Similarly, for programs identified as 'service submodule', the build outputs can be published via the package for static link scenarios. 'internal submodule' identifies sub modules that are statically called within the application context only and can be published to the package as well. Dynamically called modules are not in scope for publishing, they are resolved during runtime execution.",
```

Not sure we should detail the purpose of the value here, since it's documented already.

```suggestion
                                    "description": "The usage of the file impacts how build outputs for the file are packaged. Case-sensitive. More information on this documentation section: https://github.com/IBM/dbb-git-migration-modeler/blob/main/AdvancedInformation.md#the-assessment-phase",
```

---

I would like to combine the two. Having some details and a reference to the doc is both worth it.